doc/poe/poe_hld.md

```diff
@@ -0,0 +1,370 @@
+# SONiC PoE #
+
+## Table of Content 
+
+- [Revision](#revision)
+- [Scope](#scope)
+- [Definition/Abbreviation](#definitions_abbreviations)
+- [Overview](#overview)
+- [Requirements](#requirements)
+- [Architecture Design](#architecture-design)
+  - [PoE interaction](#poe-interaction)
+  - [SWSS changes](#swss-changes)
+  - [PoE SYNCD](#poe-syncd)
+  - [PoE Manager](#poe-manager)
+  - [LLDP Manager](#lldp-manager-modified)
+- [High-Level Design](#high-level-design)
+  - [PoE flows](#poe-flows)
+    - [PoE data collection](#poe-data-collection)
+    - [PoE configuration](#poe-configuration)
+- [SAI API](#sai-api)
+- [PoE AI](#poe-abstraction-interface)
+- [Configuration and management](#configuration-and-management)
+  - [CLI/YANG model Enhancements](#cli_yang-model-enhancements)
+    - [Show commands](#show-commands)
+    - [Config commands](#config-commands)
+  - [Config DB Enhancements](#config-db-enhancements)
+  - [State DB Enhancements](#state-db-enhancements)
+- [Restrictions/Limitations](#restrictions_limitations)
+
+### Revision
+
+ | Rev | Date |     Author     | Change Description |
+ |:---:|:----:|:--------------:|--------------------|
+ | 0.1 | 03-26-2024 | Volodymyr Mytnyk |  Initial version |
+ | 0.2 | 04-01-2024 | Volodymyr Mytnyk |  Add SWSS cahnges, include LLDP interaction, add example PoE config, mention about warm reboot requirements |
+
+### Scope  
+
+This document provides general information about the PoE (Power over Ethernet) feature implementation in SONiC.
+
+### Definitions/Abbreviations 
+
+|                          |                                |
+|--------------------------|--------------------------------|
+| PoE                      | Power over Ethernet |
+
+### Overview 
+
+Ethernet switch (e.g. campus specific platforms) today often comes with Power over Ethernet (PoE) feature support. PoE functioalility itself is considered as an external functionality from NPU SAI/SDK point of view as it is configurable via external PoE controller.
+
+
+### Requirements
+
+- PoE feature can be enabled via compile option (disabled by default);
+- PoE config persistence during SONiC reboot;
+- Support CLI configurations as mentioned in [config section](#config-commands) and [show section](#show-commands);
+- LLDP support (PoE power reporting only, future releases);
+- Warm boot;
+
+### Architecture Design 
+
+In order to isolate PoE functionality and complexity from existing SONiC design, the following main changes are made:
+
+- Add separate container for PoE control plane routine configuration.
```

Comment:
How this design address the platform dependencies?
Since there is a separate docker/container for the PoE, How this can resolve the platform dependencies like the dynamic detection of PSU and their power budget etc.